### PR TITLE
feat: Focus search input when double-tapping bottom navigation Search item

### PR DESCRIPTION
**Pull request by olerichter00**
### Description
This PR adds common app behavior and focuses on the search input (and opens the keyboard) when double-tapping the bottom navigation search item.

https://github.com/artsy/eigen/assets/4691889/5a487207-3932-421e-9dcb-c42f26db8df0

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I hid my changes behind a **[feature flag]**, or they don't need one.
- [ ] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Focus search input when double-tapping bottom navigation Search item - ole

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 53d94a965be7baee027f55f49fd856851f9ed489
</p>

File: src/app/routes.ts [hunk 1]
```diff
@@ -251,7 +251,7 @@ function getDomainMap(): Record<string, RouteMatcher[] | null> {
     addRoute("/sell/inquiry", "ConsignmentInquiry"),
     addRoute("/selling-with-artsy", "MyCollectionSellingWithartsyFAQ"),
     addRoute("/settings/alerts", "SavedSearchAlertsList"),
-    addRoute("/settings/alerts/:savedSearchAlertId", "EditSavedSearchAlert"),
+    addRoute("/settings/alerts/:savedSearchAlertId/edit", "EditSavedSearchAlert"),
```
Comment (gkartalis) [inline src/app/routes.ts:254]: question: do we need this for this feature?

File: ios/Podfile.lock [hunk 2]
```diff
@@ -1208,7 +1208,7 @@ SPEC CHECKSUMS:
   fmt: ff9d55029c625d3757ed641535fd4a75fedc7ce9
   Forgeries: 64ced144ea8341d89a7eec9d1d7986f0f1366250
   FXBlurView: db786c2561cb49a09ae98407f52460096ab8a44f
-  glog: 04b94705f318337d7ead9e6d17c019bd9b1f6b1b
+  glog: 5337263514dd6f09803962437687240c5dc39aa4
```
Comment (gkartalis) [inline ios/Podfile.lock:1211]: question: Can you run locally `cd ios && bundle exec pod update glog` to see if this is still locally present in your machine?

File: src/app/Scenes/Search/Search.tsx [hunk 3]
```diff
@@ -54,6 +56,20 @@ export const Search: React.FC = () => {
...
+    searchProviderValues.inputRef.current?.focus()
+  }
+
+  useEffect(() => {
+    BottomTabsEvents.addListener(`${SCROLL_TO_TOP_EVENT}-search`, focusSearchInput)
+
+    return () => {
+      BottomTabsEvents.removeListener(`${SCROLL_TO_TOP_EVENT}-search`, focusSearchInput)
+    }
+  }, [])
```
Comment (gkartalis) [inline src/app/Scenes/Search/Search.tsx:71]: 👏

File: src/app/routes.ts (hunk 1 above)
Comment (olerichter00) [reply src/app/routes.ts:254]: We've already [changed the links](https://github.com/artsy/eigen/blob/53d94a965be7baee027f55f49fd856851f9ed489/src/app/Scenes/SavedSearchAlertsList/Components/SavedSearchesList.tsx#L83) but forgot to update the route.

Comment (olerichter00) [reply src/app/routes.ts]: Oh, it's part of this PR https://github.com/artsy/eigen/pull/9597

I've removed the change.

File: ios/Podfile.lock (hunk 2 above)
Comment (olerichter00) [reply ios/Podfile.lock]: Thanks, I've removed the change.

**Review by MounirDhahri — DISMISSED** [review]
Nice

File: src/app/Scenes/Search/Search.tsx (hunk 3 above)
Comment (olerichter00) [reply src/app/Scenes/Search/Search.tsx:71]: No need for this effect, I've changed the code and use `useBottomTabsScrollToTop`'s [onScrollToTop](https://github.com/artsy/eigen/blob/main/src/app/utils/bottomTabsHelper.tsx#L25) callback instead.

**Review by MrSltun — DISMISSED** [review]
Great work Ole!! 🚀🚀

**Comment by damassi** [comment]
Nicee

**Pull request merged**